surfactant/cmd/createconfig.py

```diff
@@ -0,0 +1,44 @@
+import json
+from pathlib import Path
+
+import click
+from loguru import logger
+
+
+@click.command()
+@click.argument("directory", type=click.Path(exists=True))
+@click.option(
+    "-o",
+    "--output",
+    type=str,
+    help="Output JSON file name, defaults to end of directory string passed as input",
+)
+@click.option(
+    "--install-prefix",
+    type=str,
+    default="/",
+    help="Install prefix to use in the configuration (default: '/')",
+)
+def create_config(directory, output, install_prefix):
+    """Create surfactant input configuration file based on input directory."""
+    starting_directory = Path(directory)
+    extract_paths = []
+
+    # Check if there are any files in the directory
+    if any(item.is_file() for item in starting_directory.iterdir()):
+        extract_paths.append(starting_directory.as_posix())
+    else:
+        # If there are no files, add the immediate subdirectories
+        for item in starting_directory.iterdir():
+            if item.is_dir():
+                extract_paths.append(item.as_posix())
```

Comment:
A few questions that came up when I was testing with this set up:

```
.
└── a
    ├── b
    │   └── ls1
    └── c
        └── ls2
```

When I do create-config `a`, I get a config file with `a/b` and `a/c` as the extract paths -- the result when generating an sbom is that any files under both of those extract paths will appear to be installed under the same install prefix (`/ls1` and `/ls2`).

On the other hand, if a file exists as an immediate child of `a`, the extract path will be `a`, and the resulting install paths will be `/b/ls1` and `/c/ls2`.

1. Is that the desired behavior, or would it be more intuitive to have the extract path be `a` regardless of the presence of a file?
2. Would it make sense to use absolute paths in the list of extract paths? The config file wouldn't be portable between different computers, but then `surfactant generate` wouldn't need to be run from the same folder as where the config file got generated.

Reply:
It was the desired behavior but I modified to just use the input directory as that makes things more intuitive. Having it use the full paths enables surfactant to be run from any location on the system without issues. With how easy it is is create the config it shouldn't be an issue that its not portable between computers.